core/app/models/spree/address.rb

```diff
@@ -177,6 +168,19 @@ def country_iso
       country && country.iso
     end
 
+    # @return [String] the full name on this address
+    def name
+      Spree::Name.new(firstname, lastname).full_name
+    end
+
+    def name=(value)
+      return if value.nil?
+
+      current_name = Spree::Name.new(value)
```

Comment:
`current_name`: we can probably find a better variable name here. I feel like `current` has not a contextual meaning but I can't find a better one. 🤔

Reply:
It's not perfect but I can't come up we something better either

Reply:
`name_from_value` could be the best one 😉